File: CMModule/src/main/java/com/cm/app/service/LoanApplicationService.java
package com.cm.app.service;

import java.util.List;

import com.cm.app.model.LoanApplication;

public interface LoanApplicationService {

	LoanApplication createLoanApplication(LoanApplication loanApplication);

	LoanApplication getLoanApplicationById(int id);

	List<LoanApplication> getAllLoanApplications();

	LoanApplication updateLoanApplication(int id, LoanApplication loanApplication);

	List<LoanApplication> getByStatus(String status);
}

File: homeloanfrontend/src/app/cm/loan-application.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { LoanApplication } from './model/LoanApplication';

@Injectable({
  providedIn: 'root'
})
export class LoanApplicationService {

  private baseUrl = 'http://localhost:4000/loan-applications';

  constructor(private http: HttpClient) { }

  createLoanApplication(loanApplication: LoanApplication): Observable<LoanApplication> {
    return this.http.post<LoanApplication>(this.baseUrl, loanApplication);
  }

  getLoanApplicationById(id: number): Observable<LoanApplication> {
    const url = `${this.baseUrl}/${id}`;
    return this.http.get<LoanApplication>(url);
  }

  getAllLoanApplications(): Observable<LoanApplication[]> {
    return this.http.get<LoanApplication[]>(this.baseUrl);
  }

  updateLoanApplication(id: number, loanApplication: LoanApplication): Observable<LoanApplication> {
    const url = `${this.baseUrl}/${id}`;
    return this.http.put<LoanApplication>(url, loanApplication);
  }

  
}
